File: packages/transform/src/features/misc/MiscCreateAssertCloneTransformer.ts
import { MiscAssertCloneProgrammer } from "@typia/core";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace MiscCreateAssertCloneTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.factory({
      ...props,
      method: "misc.createAssertClone",
      write: MiscAssertCloneProgrammer.write,
    });
}

File: packages/transform/src/features/misc/MiscCreateAssertCloneTransformer.go
package misc

import internal "github.com/samchon/typia/packages/transform/src/internal"

var MiscCreateAssertCloneTransformer = miscCreateAssertCloneTransformerNamespace{}

type miscCreateAssertCloneTransformerNamespace struct{}

func (miscCreateAssertCloneTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return "(input => JSON.parse(JSON.stringify(input)))"
}
